diff --git a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/applications.rnc b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/applications.rnc
index c45417cc..b7491c97 100644
--- a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/applications.rnc
+++ b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/applications.rnc
@@ -1,4 +1,4 @@
-datatypes w = "http://whattf.org/datatype-draft"
+
 
 # #####################################################################
 ##  RELAX NG Schema for HTML 5: Web Application Features              #
@@ -20,40 +20,40 @@ datatypes w = "http://whattf.org/datatype-draft"
 
 	common.attrs.contenteditable =
 		attribute contenteditable {
-			w:string "true" | w:string "false" | w:string ""
+			 "true" |  "false" |  ""
 		}
 
 ## Draggable Element: draggable
 
 	common.attrs.draggable =
 		attribute draggable {
-			w:string "true" | w:string "false"
+			 "true" |  "false"
 		}
 
 ## Hidden Element: hidden
 
 	common.attrs.hidden =
 		attribute hidden {
-			w:string "hidden" | w:string ""
+			 "hidden" |  ""
 		}
 
 ## Spellchecking and grammar checking: spellcheck
 
 	common.attrs.spellcheck =
 		attribute spellcheck{
-			w:string "true" | w:string "false" | w:string ""
+			 "true" |  "false" |  ""
 		}
 
 ## Autocapitalization
 
 	common.attrs.autocapitalize =
 		attribute autocapitalize {
-			(	w:string "off"
-			|	w:string "none"
-			|	w:string "on"
-			|	w:string "sentences"
-			|	w:string "words"
-			|	w:string "characters"
+			(	 "off"
+			|	 "none"
+			|	 "on"
+			|	 "sentences"
+			|	 "words"
+			|	 "characters"
 			)
 		}
 
@@ -114,7 +114,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		dialog.attrs.open =
 			attribute open {
-				w:string "open" | w:string ""
+				 "open" |  ""
 			}
 	dialog.inner =
 		( common.inner.flow )
@@ -205,7 +205,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		details.attrs.open =
 			attribute open {
-				w:string "open" | w:string ""
+				 "open" |  ""
 			}
 	details.inner =
 		(	summary.elem
diff --git a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/block.rnc b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/block.rnc
index a6afee23..e2fc7485 100644
--- a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/block.rnc
+++ b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/block.rnc
@@ -1,4 +1,3 @@
-datatypes w = "http://whattf.org/datatype-draft"
 
 # #####################################################################
 ##  RELAX NG Schema for HTML 5: Basic Prose Markup                    #
@@ -137,11 +136,11 @@ datatypes w = "http://whattf.org/datatype-draft"
 			}
 		ol.attrs.reversed =
 			attribute reversed {
-				w:string "reversed" | w:string ""
+				 "reversed" |  ""
 			}
 		ol.attrs.type =
 			attribute type {
-				w:string "1" | w:string "a" | w:string "A" | w:string "i" | w:string "I"
+				 "1" |  "a" |  "A" |  "i" |  "I"
 			}
 	ol.inner =
 		(	oli.elem*
diff --git a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/common.rnc b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/common.rnc
index 4f18cfee..d208352e 100644
--- a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/common.rnc
+++ b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/common.rnc
@@ -1,4 +1,3 @@
-datatypes w = "http://whattf.org/datatype-draft"
 
 # #####################################################################
 ##  RELAX NG Schema for HTML 5: Common Definitions                    #
@@ -147,7 +146,7 @@ common.attrs.i18n =
 	)
 	common.attrs.dir =
 		attribute dir {
-			w:string "ltr" | w:string "rtl" | w:string "auto"
+			 "ltr" |  "rtl" |  "auto"
 		}
 	# This lang definition is a hack for environments where
 	# the HTML5 parser maps lang to xml:lang.
@@ -178,7 +177,7 @@ common.attrs.i18n =
 #		} & XMLonly
 	common.attrs.translate =
 		attribute translate {
-			w:string "" | w:string "yes" | w:string "no"
+			 "" |  "yes" |  "no"
 		}
 
 common.attrs.present =
@@ -213,10 +212,10 @@ common.attrs.other =
 		list { token* }
 
 	common.data.browsing-context =
-		w:browsing-context
+		datatype.html5.browsing.context.name
 
 	common.data.browsing-context-or-keyword =
-		w:browsing-context-or-keyword
+		datatype.html5.browsing.context.name.or.keyword
 #		xsd:string {
 #			pattern = "()|([^_].*)|(_[bB][lL][aA][nN][kK])|(_[sS][eE][lL][fF])|(_[pP][aA][rR][eE][nN][tT])|(_[tT][oO][pP])"
 #		}
@@ -225,22 +224,22 @@ common.attrs.other =
 ## IDs and IDREFs
 
 	common.data.id =
-		w:ID
+		datatype.html5.token
 #		xsd:string {
 #			pattern = "\S+"
 #		}
 
 	common.data.idref =
-		w:IDREF
+		datatype.html5.token
 		
 	common.data.idrefs =
-		w:IDREFS
+		datatype.html5.space.separated.tokens
 
 	common.data.name =
-		w:ID #FIXME
+		datatype.html5.token
 
 	common.data.hash-name =
-		w:hash-name	
+		datatype.html5.name.reference	
 #		xsd:string {
 #			pattern = "#.+"
 #		}
@@ -248,19 +247,19 @@ common.attrs.other =
 ## Numerical
 
 	common.data.integer =
-		w:integer
+		datatype.integer
 #		xsd:string {
 #			pattern = "-?[0-9]+"
 #		}
 
 	common.data.integer.positive =
-		w:integer-positive
+		datatype.positive.integer
 #		xsd:string {
 #			pattern = "0*[1-9][0-9]*"
 #		}
 
 	common.data.integer.non-negative =
-		w:integer-non-negative
+		datatype.non-negative.integer
 #		xsd:string {
 #			pattern = "[0-9]+"
 #		}
@@ -271,19 +270,19 @@ common.attrs.other =
 #		}
 		
 	common.data.float =
-		w:float
+		datatype.float
 #		xsd:string {
 #			pattern = "-?[0-9]+(\.[0-9]+)?([eE]-?[0-9]+)?"
 #		}
 
 	common.data.float.positive =
-		w:float-positive
+		datatype.float.positive
 #		xsd:string {
 #			pattern = "(0*[1-9][0-9]*(\.[0-9]+)?)|([0-9]+(\.0*[1-9][0-9]*)?)([eE]-?[0-9]+)?"
 #		}
 		
 	common.data.float.non-negative =
-		w:float-non-negative
+		datatype.float.non-negative
 #		xsd:string {
 #			pattern = "[0-9]+(\.[0-9]+)?([eE]-?[0-9]+)?"
 #		}
@@ -291,20 +290,20 @@ common.attrs.other =
 ## Temporal
 
 	common.data.datetime =
-		w:datetime-tz
+		datatype.html5.datetime
 
 	common.data.date =
-		w:date
+		datatype.html5.date
 
 	common.data.time-datetime =
-		w:time-datetime
+		datatype.html5.date.or.time
 
 ## IRIs
 
 	common.data.uri =
 		# allow either a non-empty IRI ref or zero or more HTML
 		# space characters (which are: space, tab, LF, FF, CR)
-		(	w:iri-ref
+		(	datatype.IRI
 		|	xsd:string {
 				pattern = "[ \x{0A}-\x{0D}]*"
 				# NOTE The range above incorrectly allows U+000B
@@ -317,16 +316,16 @@ common.attrs.other =
 		)
 
 	common.data.uri.non-empty =
-		w:iri-ref
+		datatype.IRI
 
 	common.data.uris =
-		list { w:iri-ref* }
+		list { datatype.IRI* }
 
 	common.data.uris.http.or.https =
-		list { w:iri-ref-http-or-https* }
+		list { datatype.URI.absolute* }
 
 	common.data.uri.absolute =
-		w:iri
+		datatype.URI.absolute
 
 ## <link type='icon'> sizes
 
@@ -341,30 +340,30 @@ common.attrs.other =
 ## color for <link rel="mask-icon"> and <meta name="theme-color">
 
 	common.data.color =
-		w:color
+		datatype.html5.color
 
 ## MIME types
 
 	common.data.mimetype =
-		w:mime-type
+		datatype.mimetype
 
 ## Script type
 
 	common.data.scripttype =
-		w:script-type
+		datatype.string
 
 ## Encodings
 
 	common.data.charset =
-		w:charset
+		datatype.character.encoding
 
 	common.data.meta-charset =
-		w:meta-charset
+		datatype.string
 
 ## Refresh
 
 	common.data.refresh =
-		w:refresh
+		datatype.string
 
 ## Default style
 
@@ -380,28 +379,28 @@ common.attrs.other =
 ## Media Queries
 
 	common.data.mediaquery =
-		w:media-query
+		datatype.html5.media.query
 
 ## Language Codes
 
 	common.data.langcode =
-		w:string "" | w:language
+		 "" | datatype.languagecode
 
 ## List of Key Labels
 	common.data.keylabellist =
-		w:keylabellist
+		datatype.html5.accesskey
 
 ## List of Source Sizes
 	common.data.source.size.list =
-		w:source-size-list
+		datatype.string
 
 ## Subresource Integrity
 	common.data.integrity =
-		w:integrity-metadata
+		datatype.string
 
 ## Content Security Policy
 	common.data.content-security-policy =
-		w:content-security-policy
+		datatype.string
 
 ## Feature Policy
 	common.data.feature-policy =
@@ -409,19 +408,20 @@ common.attrs.other =
 
 ## List of sandbox keywords
 	common.data.sandbox-allow-list =
-		w:string "" | w:sandbox-allow-list
+		 "" | list { 'allow-top-navigation'?, 'allow-same-origin'?, 'allow-forms'?, 'allow-scripts'? }
+        | list { 'allow-top-navigation'?, 'allow-same-origin'?, 'allow-scripts'?, 'allow-forms'? }
 
 ## Custom element name
 	common.data.custom.element.name =
-		w:custom-element-name
+		datatype.string
 
 ## Microdata Properties
   common.data.microdata-properties =
-    list { w:microdata-property+ }
+    list { datatype.string+ }
 
 ## Zero
   common.data.zero =
-    w:zero
+    "0"
 
 ## ECMAScript FunctionBody
   common.data.functionbody =
@@ -430,7 +430,7 @@ common.attrs.other =
 ## autocomplete
 
 	common.data.autocomplete.any =
-		w:autocomplete-any
+		datatype.string
 
 # #####################################################################
 ##  WF2 Module Hook                                                   #
diff --git a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/core-scripting.rnc b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/core-scripting.rnc
index 2cf2848f..cac25cbd 100644
--- a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/core-scripting.rnc
+++ b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/core-scripting.rnc
@@ -1,4 +1,3 @@
-datatypes w = "http://whattf.org/datatype-draft"
 
 # #####################################################################
 ##  RELAX NG Schema for HTML 5: Core Scripting                        #
@@ -47,19 +46,19 @@ datatypes w = "http://whattf.org/datatype-draft"
 			}
 		script.attrs.defer =
 			attribute defer {
-				w:string "defer" | w:string ""
+				 "defer" |  ""
 			}
 		script.attrs.async =
 			attribute async {
-				w:string "async" | w:string ""
+				 "async" |  ""
 			} & v5only
 		script.attrs.type =
 			attribute type {
-				w:string "module" | w:string "" | common.data.scripttype # XXX without charset parameter!
+				 "module" |  "" | common.data.scripttype # XXX without charset parameter!
 			}
 		script.attrs.nomodule =
 			attribute nomodule {
-				w:string "nomodule" | w:string ""
+				 "nomodule" |  ""
 			}
 		script.attrs.charset =
 			attribute charset {
diff --git a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/data.rnc b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/data.rnc
index 24c2b110..3878e65a 100644
--- a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/data.rnc
+++ b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/data.rnc
@@ -1,4 +1,3 @@
-datatypes w = "http://whattf.org/datatype-draft"
 
 # #####################################################################
 ##  RELAX NG Schema for HTML 5: Static Data Markup                    #
diff --git a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/embed.rnc b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/embed.rnc
index ce8f38c3..3f5e59f4 100644
--- a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/embed.rnc
+++ b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/embed.rnc
@@ -1,4 +1,3 @@
-datatypes w = "http://whattf.org/datatype-draft"
 namespace local = ""
 
 # #####################################################################
@@ -60,7 +59,7 @@ namespace local = ""
 			}
 		img.attrs.ismap =
 			attribute ismap {
-				w:string "ismap" | w:string ""
+				 "ismap" |  ""
 			}
 		img.attrs.decoding =
 			attribute decoding {
@@ -380,7 +377,7 @@ namespace local = ""
 			}
 		object.attrs.typemustmatch =
 			attribute typemustmatch {
-				w:string "typemustmatch" | w:string ""
+				 "typemustmatch" |  ""
 			} & v5only
 		object.attrs.height =
 			attribute height {
@@ -482,15 +479,15 @@ namespace local = ""
 			} & v5only
 		iframe.attrs.allowfullscreen =
 			attribute allowfullscreen {
-				w:string "allowfullscreen" | w:string ""
+				 "allowfullscreen" |  ""
 			} & v5only
 		iframe.attrs.allowpaymentrequest =
 			attribute allowpaymentrequest {
-				w:string "allowpaymentrequest" | w:string ""
+				 "allowpaymentrequest" |  ""
 			} & v5only
 		iframe.attrs.allowusermedia =
 			attribute allowusermedia {
-				w:string "allowusermedia" | w:string ""
+				 "allowusermedia" |  ""
 			} & v5only
 		iframe.attrs.allow =
 			attribute allow {
@@ -564,40 +561,40 @@ namespace local = ""
 			)
 		area.attrs.shape.rect =
 			attribute shape {
-				w:string "rect"
+				 "rect"
 			}
 		area.attrs.coords.rect =
 			attribute coords {
-				w:rectangle
+				datatype.html5.coords.rectangle
 #				xsd:token {
 #					pattern = "-?[0-9]+,-?[0-9]+,-?[0-9]+,-?[0-9]+"
 #				}
 			}
 		area.attrs.shape.circle =
 			attribute shape {
-				w:string "circle"
+				 "circle"
 			}
 		area.attrs.coords.circle =
 			attribute coords {
-				w:circle
+				datatype.html5.coords.circle
 #				xsd:token {
 #					pattern = "-?[0-9]+,-?[0-9]+,[0-9]+"
 #				}
 			}
 		area.attrs.shape.poly =
 			attribute shape {
-				w:string "poly"
+				 "poly"
 			}
 		area.attrs.coords.poly =
 			attribute coords {
-				w:polyline
+				datatype.html5.coords.poly
 #				xsd:token {
 #					pattern = "-?[0-9]+,-?[0-9]+,-?[0-9]+,-?[0-9]+,-?[0-9]+,-?[0-9]+(,-?[0-9]+,-?[0-9]+)*"
 #				}
 			}
 		area.attrs.shape.default =
 			attribute shape {
-				w:string "default"
+				 "default"
 			}
 	area.inner =
 		( empty )
@@ -608,5 +605,5 @@ namespace local = ""
 
 		embedded.content.attrs.crossorigin =
 			attribute crossorigin {
-				w:string "anonymous" | w:string "use-credentials" | w:string ""
+				 "anonymous" |  "use-credentials" |  ""
 			} & v5only
diff --git a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/form-datatypes.rnc b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/form-datatypes.rnc
index 76fa0567..48d83222 100644
--- a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/form-datatypes.rnc
+++ b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/form-datatypes.rnc
@@ -1,4 +1,3 @@
-datatypes w = "http://whattf.org/datatype-draft"
 
 # #####################################################################
 ##  RELAX NG Schema for HTML 5: Datatypes related to forms            #
@@ -7,7 +6,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 ## MIME types
 		
 	form.data.mimetypelist = 
-		w:mime-type-list
+		datatype.html5.forms.mimetypelist
 		
 
 ## ECMAScript Regular Expression
@@ -18,32 +17,32 @@ datatypes w = "http://whattf.org/datatype-draft"
 ## Temporal
 		
 	form.data.datetime-local =
-		w:datetime-local
+		datatype.html5.datetime.local
 	
 	form.data.date =
-		w:date
+		datatype.html5.date
 	
 	form.data.month =
-		w:month
+		datatype.html5.month
 	
 	form.data.week =
-		w:week
+		datatype.html5.week 
 	
 	form.data.time =
-		w:time
+		datatype.html5.time
 
 ## Email
 
 	form.data.emailaddress =
-		w:email-address
+		datatype.html5.emailaddress
 
 	form.data.emailaddresslist =
-		w:email-address-list
+		datatype.html5.emailaddresslist
 
 ## Color
 
 	form.data.color =
-		w:simple-color
+		datatype.html5.color
 #		xsd:string {
 #			pattern = "#[a-fA-F0-9]{6}"
 #		}
@@ -51,10 +50,10 @@ datatypes w = "http://whattf.org/datatype-draft"
 ## Text without line breaks
 
 	form.data.stringwithoutlinebreaks =
-		w:string-without-line-breaks
+		datatype.html5.forms.string.nolinebreaks
 
 ## Non-empty string
 
 	form.data.nonemptystring =
-		w:non-empty-string
+		datatype.string.nonempty
 
diff --git a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/media.rnc b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/media.rnc
index 5dd1b743..ae032055 100644
--- a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/media.rnc
+++ b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/media.rnc
@@ -17,23 +17,23 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		media.attrs.autoplay   =
 			attribute autoplay   {
-				w:string "autoplay" | w:string ""
+				 "autoplay" |  ""
 			}
 		media.attrs.preload   =
 			attribute preload   {
-				w:string "none" | w:string "metadata" | w:string "auto" | w:string ""
+				 "none" |  "metadata" |  "auto" |  ""
 			}
 		media.attrs.controls  =
 			attribute controls  {
-				w:string "controls" | w:string ""
+				 "controls" |  ""
 			}
 		media.attrs.loop =
 			attribute loop {
-				w:string "loop" | w:string ""
+				 "loop" |  ""
 			}
 		media.attrs.muted =
 			attribute muted {
-				w:string "muted" | w:string ""
+				 "muted" |  ""
 			}
 		
 ## Source: <source>
@@ -100,7 +100,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 			}
 		video.attrs.playsinline =
 			attribute playsinline {
-				w:string "playsinline" | w:string ""
+				 "playsinline" |  ""
 			}
 	video.inner.flow =
 		(	media.source
@@ -159,7 +159,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		track.attrs.kind =
 			attribute kind {
-				w:string "subtitles" | w:string "captions" | w:string "descriptions" | w:string "chapters" | w:string "metadata"
+				 "subtitles" |  "captions" |  "descriptions" |  "chapters" |  "metadata"
 			}
 		track.attrs.src =
 			attribute src {
@@ -175,7 +175,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 			}
 		track.attrs.default =
 			attribute default {
-				w:string "default" | w:string ""
+				 "default" |  ""
 			}
 
 	track.inner =
diff --git a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/meta.rnc b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/meta.rnc
index fb247926..45ce4123 100644
--- a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/meta.rnc
+++ b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/meta.rnc
@@ -1,4 +1,3 @@
-datatypes w = "http://whattf.org/datatype-draft"
 
 # #####################################################################
 ##  RELAX NG Schema for HTML 5: Global Structure & Metadata          #
@@ -176,7 +175,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 			}
 		link.attrs.rel =
 			attribute rel {
-				w:link-rel
+				datatype.html5.space.separated.tokens
 			}
 		link.attrs.as =
 			attribute as {
@@ -210,7 +209,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 			}
 		link.attrs.sizes =
 			attribute sizes {
-				w:string "any" | common.data.sizes
+				 "any" | common.data.sizes
 			}
 		link.attrs.color =
 			attribute color {
@@ -290,7 +289,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		meta.name.attrs.name =
 			attribute name {
-				w:non-empty-string
+				datatype.string.nonempty
 			}
 		meta.name.attrs.content =
 			attribute content {
@@ -318,7 +317,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		meta.http-equiv.attrs.http-equiv.refresh =
 			attribute http-equiv {
-				w:string "refresh"
+				 "refresh"
 			}
 		meta.http-equiv.attrs.content.refresh =
 			attribute content { 
@@ -345,7 +344,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		meta.http-equiv.attrs.http-equiv.default-style =
 			attribute http-equiv {
-				w:string "default-style"
+				 "default-style"
 			}
 		meta.http-equiv.attrs.content.default-style =
 			attribute content {
@@ -373,7 +372,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		meta.http-equiv.attrs.http-equiv.content-security-policy =
 			attribute http-equiv {
-				w:string "content-security-policy"
+				 "content-security-policy"
 			}
 		meta.http-equiv.attrs.content.content-security-policy =
 			attribute content {
@@ -398,7 +397,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		meta.http-equiv.attrs.http-equiv.x-ua-compatible =
 			attribute http-equiv {
-				w:string "x-ua-compatible"
+				 "x-ua-compatible"
 			}
 		meta.http-equiv.attrs.content.x-ua-compatible =
 			attribute content {
@@ -446,7 +445,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		meta.http-equiv.attrs.http-equiv.content-type =
 			attribute http-equiv {
-				w:string "content-type"
+				 "content-type"
 			}
 		meta.http-equiv.attrs.content.content-type =
 			attribute content {
diff --git a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/microdata.rnc b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/microdata.rnc
index 91616c84..21ca2a36 100644
--- a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/microdata.rnc
+++ b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/microdata.rnc
@@ -1,4 +1,3 @@
-datatypes w = "http://whattf.org/datatype-draft"
 
 # #####################################################################
 ##  RELAX NG Schema for HTML 5: Microdata                             #
@@ -21,7 +20,7 @@ common.attrs.microdata =
 		}
 	common.attrs.microdata.itemscope =
 		attribute itemscope {
-			w:string "itemscope" | w:string ""
+			 "itemscope" |  ""
 		}
 	common.attrs.microdata.itemtype =
 		attribute itemtype {
diff --git a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/phrase.rnc b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/phrase.rnc
index 0ebabf8f..22ec1531 100644
--- a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/phrase.rnc
+++ b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/phrase.rnc
@@ -1,4 +1,3 @@
-datatypes w = "http://whattf.org/datatype-draft"
 
 # #####################################################################
 ##  RELAX NG Schema for HTML 5: Phrase Markup                         #
@@ -70,7 +69,7 @@ datatypes w = "http://whattf.org/datatype-draft"
     }
   shared-hyperlink.attrs.rel =
     attribute rel {
-      w:a-rel
+      datatype.html5.space.separated.tokens
     }
   shared-hyperlink.attrs.hreflang =
     attribute hreflang {
diff --git a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/ruby.rnc b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/ruby.rnc
index e58992ba..91f3b84b 100644
--- a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/ruby.rnc
+++ b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/ruby.rnc
@@ -1,4 +1,3 @@
-datatypes w = "http://whattf.org/datatype-draft"
 namespace local = ""
 
 # #####################################################################
diff --git a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/tables.rnc b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/tables.rnc
index c5c62e2b..e256366f 100644
--- a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/tables.rnc
+++ b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/tables.rnc
@@ -1,4 +1,3 @@
-datatypes w = "http://whattf.org/datatype-draft"
 
 # #####################################################################
 ##  RELAX NG Schema for HTML 5: Tables                                #
@@ -24,11 +23,11 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		tables.attrs.align =
 			attribute align {
-				(	w:string "left" 
-				|	w:string "center"
-				|	w:string "right"
-				|	w:string "justify"
-				|	w:string "char"
+				(	 "left" 
+				|	 "center"
+				|	 "right"
+				|	 "justify"
+				|	 "char"
 				)
 			}
 		tables.attrs.char =
@@ -37,10 +36,10 @@ datatypes w = "http://whattf.org/datatype-draft"
 			}
 		tables.attrs.valign =
 			attribute valign {
-				(	w:string "top"
-				|	w:string "middle"
-				|	w:string "bottom"
-				|	w:string "baseline"
+				(	 "top"
+				|	 "middle"
+				|	 "bottom"
+				|	 "baseline"
 				)
 			}
 
@@ -200,10 +199,10 @@ datatypes w = "http://whattf.org/datatype-draft"
 		( tables.attrs.scope? )
 		tables.attrs.scope =
 			attribute scope {
-				(	w:string "row"
-				|	w:string "col"
-				|	w:string "rowgroup"
-				|	w:string "colgroup"
+				(	 "row"
+				|	 "col"
+				|	 "rowgroup"
+				|	 "colgroup"
 				)
 			}
 		tables.attrs.abbr =
diff --git a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/web-components.rnc b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/web-components.rnc
index 5cb3d5d1..be8b8c01 100644
--- a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/web-components.rnc
+++ b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/web-components.rnc
@@ -1,4 +1,3 @@
-datatypes w = "http://whattf.org/datatype-draft"
 namespace c = "http://n.validator.nu/custom-elements/"
 
 # #####################################################################
diff --git a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/web-forms.rnc b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/web-forms.rnc
index 9889d783..cfdf617f 100644
--- a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/web-forms.rnc
+++ b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/web-forms.rnc
@@ -1,4 +1,3 @@
-datatypes w = "http://whattf.org/datatype-draft"
 
 # #####################################################################
 ##  RELAX NG Schema for HTML 5: Web Forms 1.0 markup                  #
@@ -18,12 +17,12 @@ datatypes w = "http://whattf.org/datatype-draft"
 
 	common-form.attrs.disabled = 
 		attribute disabled {
-			w:string "disabled" | w:string ""
+			 "disabled" |  ""
 		}
 
 	shared-form.attrs.readonly = 
 		attribute readonly {
-			w:string "readonly" | w:string ""
+			 "readonly" |  ""
 		}
 
 	shared-form.attrs.maxlength = 
@@ -42,7 +41,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 	
 	input.attrs.checked = 
 		attribute checked {
-			w:string "checked" | w:string ""
+			 "checked" |  ""
 		}
 
 ## Text Field: <input type='text'>
@@ -66,7 +65,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		input.text.attrs.type = 
 			attribute type {
-				w:string "text"
+				 "text"
 			}
 		input.text.attrs.value =
 			attribute value {
@@ -90,7 +89,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		input.password.attrs.type = 
 			attribute type {
-				w:string "password"
+				 "password"
 			}
 		input.password.attrs.value =
 			attribute value {
@@ -119,7 +118,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		input.checkbox.attrs.type = 
 			attribute type {
-				w:string "checkbox"
+				 "checkbox"
 			}
 		input.checkbox.attrs.value =
 			attribute value {
@@ -145,7 +144,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		input.radio.attrs.type = 
 			attribute type {
-				w:string "radio"
+				 "radio"
 			}
 		input.radio.attrs.value =
 			attribute value {
@@ -176,7 +175,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		input.button.attrs.type = 
 			attribute type {
-				w:string "button"
+				 "button"
 			}
 		input.button.attrs.value =
 			attribute value {
@@ -201,7 +200,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		input.submit.attrs.type = 
 			attribute type {
-				w:string "submit"
+				 "submit"
 			}
 		input.submit.attrs.value =
 			attribute value {
@@ -225,7 +224,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		input.reset.attrs.type = 
 			attribute type {
-				w:string "reset"
+				 "reset"
 			}
 		input.reset.attrs.value =
 			attribute value {
@@ -250,7 +249,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		input.file.attrs.type = 
 			attribute type {
-				w:string "file"
+				 "file"
 			}
 		input.file.attrs.accept = 
 			attribute accept {
@@ -271,7 +270,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		input.hidden.attrs.type = 
 			attribute type {
-				w:string "hidden"
+				 "hidden"
 			}
 		input.hidden.attrs.value =
 			attribute value {
@@ -302,7 +301,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)	
 		input.image.attrs.type = 
 			attribute type {
-				w:string "image"
+				 "image"
 			}
 		input.image.attrs.alt = 
 			attribute alt {
@@ -371,7 +370,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		option.attrs.selected =
 			attribute selected {
-				w:string "selected" | w:string ""
+				 "selected" |  ""
 			}
 		option.attrs.label =
 			attribute label {
@@ -426,7 +425,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 			}
 		select.attrs.multiple =
 			attribute multiple {
-				w:string "multiple" | w:string ""
+				 "multiple" |  ""
 			}
 	select.inner =
 		(	optgroup.elem*
@@ -468,7 +467,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		button.submit.attrs.type =
 			attribute type {
-				w:string "submit"
+				 "submit"
 			}
 
 	button.elem = button.submit.elem
@@ -496,7 +495,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		button.reset.attrs.type =
 			attribute type {
-				w:string "reset"
+				 "reset"
 			}
 
 	button.elem |= button.reset.elem
@@ -525,7 +524,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		button.button.attrs.type =
 			attribute type {
-				w:string "button"
+				 "button"
 			}
 
 	button.elem |= button.button.elem
@@ -558,14 +557,14 @@ datatypes w = "http://whattf.org/datatype-draft"
 				form.attrs.method.data
 			}
 			form.attrs.method.data = 
-				( w:string "get" | w:string "post" )
+				(  "get" |  "post" )
 		form.attrs.enctype =
 			attribute enctype {
 				form.attrs.enctype.data
 			}
 			form.attrs.enctype.data = 
-				(	w:string "application/x-www-form-urlencoded" 
-				|	w:string "multipart/form-data"
+				(	 "application/x-www-form-urlencoded" 
+				|	 "multipart/form-data"
 				)
 		form.attrs.accept-charset =
 			attribute accept-charset {
diff --git a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/web-forms2.rnc b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/web-forms2.rnc
index f7ac066a..a263e315 100644
--- a/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/web-forms2.rnc
+++ b/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/web-forms2.rnc
@@ -1,4 +1,3 @@
-datatypes w = "http://whattf.org/datatype-draft"
 
 # #####################################################################
 ##  RELAX NG Schema for HTML 5: Web Forms 2.0 markup                  #
@@ -24,9 +23,9 @@ datatypes w = "http://whattf.org/datatype-draft"
 			shared-form.attrs.formenctype.data
 		}
 		shared-form.attrs.formenctype.data = 
-			(	w:string "application/x-www-form-urlencoded" 
-			|	w:string "multipart/form-data"
-			|	w:string "text/plain"
+			(	 "application/x-www-form-urlencoded" 
+			|	 "multipart/form-data"
+			|	 "text/plain"
 			)
 
 	shared-form.attrs.formmethod =
@@ -34,8 +33,8 @@ datatypes w = "http://whattf.org/datatype-draft"
 			shared-form.attrs.formmethod.data
 		}
 		shared-form.attrs.formmethod.data = 
-			( w:string "get"
-			| w:string "post"
+			(  "get"
+			|  "post"
 			)
 
 	shared-form.attrs.formtarget = 
@@ -45,12 +44,12 @@ datatypes w = "http://whattf.org/datatype-draft"
 
 	shared-form.attrs.formnovalidate = 
 		attribute formnovalidate {
-			w:string "formnovalidate" | w:string ""
+			 "formnovalidate" |  ""
 		}
 
 	shared-form.attrs.autofocus = 
 		attribute autofocus {
-			w:string "autofocus" | w:string ""
+			 "autofocus" |  ""
 		}
 	
 	shared-form.attrs.pattern = 
@@ -65,7 +64,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 
 	shared-form.attrs.required = 
 		attribute required {
-			w:string "required" | w:string ""
+			 "required" |  ""
 		}
 
 	shared-form.attrs.placeholder = 
@@ -110,24 +109,24 @@ datatypes w = "http://whattf.org/datatype-draft"
 	
 	input.attrs.step.float = 
 		attribute step {
-			w:string "any" | common.data.float.positive
+			 "any" | common.data.float.positive
 		}
 	
 	input.attrs.step.integer = 
 		attribute step {
-			w:string "any" | common.data.integer.positive 
+			 "any" | common.data.integer.positive 
 		}
 	
 	input.attrs.multiple = 
 		attribute multiple {
-			w:string "multiple" | w:string ""
+			 "multiple" |  ""
 		}
 
 ## autocomplete
 
 	input.attrs.autocomplete.any =
 		attribute autocomplete {
-			(	w:string "on" | w:string "off"
+			(	 "on" |  "off"
 			|	common.data.autocomplete.any
 			)
 		}
@@ -251,7 +250,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)	
 		input.datetime-local.attrs.type = 
 			attribute type {
-				w:string "datetime-local"
+				 "datetime-local"
 			}
 		input.datetime-local.attrs.min =
 			attribute min {
@@ -263,7 +262,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 			}	
 		input.datetime-local.attrs.value =
 			attribute value {
-				w:string "" | form.data.datetime-local
+				 "" | form.data.datetime-local
 			}
 		
 	input.elem |= input.datetime-local.elem
@@ -287,7 +286,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)	
 		input.date.attrs.type = 
 			attribute type {
-				w:string "date"
+				 "date"
 			}
 		input.date.attrs.min =
 			attribute min {
@@ -299,7 +298,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 			}	
 		input.date.attrs.value =
 			attribute value {
-				w:string "" | form.data.date
+				 "" | form.data.date
 			}
 		
 	input.elem |= input.date.elem
@@ -323,7 +322,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)	
 		input.month.attrs.type = 
 			attribute type {
-				w:string "month"
+				 "month"
 			}
 		input.month.attrs.min =
 			attribute min {
@@ -335,7 +334,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 			}	
 		input.month.attrs.value =
 			attribute value {
-				w:string "" | form.data.month
+				 "" | form.data.month
 			}
 		
 	input.elem |= input.month.elem
@@ -359,7 +358,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)	
 		input.time.attrs.type = 
 			attribute type {
-				w:string "time"
+				 "time"
 			}
 		input.time.attrs.min =
 			attribute min {
@@ -371,7 +370,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 			}	
 		input.time.attrs.value =
 			attribute value {
-				w:string "" | form.data.time
+				 "" | form.data.time
 			}
 		
 	input.elem |= input.time.elem
@@ -395,7 +394,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)	
 		input.week.attrs.type = 
 			attribute type {
-				w:string "week"
+				 "week"
 			}
 		input.week.attrs.min =
 			attribute min {
@@ -407,7 +406,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 			}	
 		input.week.attrs.value =
 			attribute value {
-				w:string "" | form.data.week
+				 "" | form.data.week
 			}
 		
 	input.elem |= input.week.elem
@@ -431,7 +430,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)	
 		input.number.attrs.type = 
 			attribute type {
-				w:string "number"
+				 "number"
 			}
 		input.number.attrs.min =
 			attribute min {
@@ -443,7 +442,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 			}	
 		input.number.attrs.value =
 			attribute value {
-				w:string "" | common.data.float
+				 "" | common.data.float
 			}
 		
 	input.elem |= input.number.elem
@@ -467,7 +466,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)	
 		input.range.attrs.type = 
 			attribute type {
-				w:string "range"
+				 "range"
 			}
 		input.range.attrs.min =
 			attribute min {
@@ -506,7 +505,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)	
 		input.email.attrs.type = 
 			attribute type {
-				w:string "email"
+				 "email"
 			}
 		input.email.attrs.value.single =
 			attribute value {
@@ -537,11 +536,11 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)	
 		input.url.attrs.type = 
 			attribute type {
-				w:string "url"
+				 "url"
 			}
 		input.url.attrs.value =
 			attribute value {
-				w:string "" | common.data.uri.absolute
+				 "" | common.data.uri.absolute
 			}
 		
 	input.elem |= input.url.elem
@@ -563,7 +562,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)	
 		input.search.attrs.type = 
 			attribute type {
-				w:string "search"
+				 "search"
 			}
 		input.search.attrs.value =
 			attribute value {
@@ -590,7 +589,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)	
 		input.tel.attrs.type = 
 			attribute type {
-				w:string "tel"
+				 "tel"
 			}
 		input.tel.attrs.value =
 			attribute value {
@@ -613,11 +612,11 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)	
 		input.color.attrs.type = 
 			attribute type {
-				w:string "color"
+				 "color"
 			}
 		input.color.attrs.value =
 			attribute value {
-				w:string "" | form.data.color
+				 "" | form.data.color
 			}
 		
 	input.elem |= input.color.elem
@@ -670,11 +669,11 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		textarea.attrs.wrap.hard =
 			attribute wrap {
-				w:string "hard"
+				 "hard"
 			}
 		textarea.attrs.wrap.soft =
 			attribute wrap {
-				w:string "soft"
+				 "soft"
 			}
 		textarea.attrs.placeholder =
 			attribute placeholder {
@@ -727,7 +726,7 @@ datatypes w = "http://whattf.org/datatype-draft"
 		)
 		form.attrs.novalidate = 
 			attribute novalidate {
-				w:string "novalidate" | w:string ""
+				 "novalidate" |  ""
 			}
 		form.attrs.target = 
 			attribute target {
@@ -735,11 +734,11 @@ datatypes w = "http://whattf.org/datatype-draft"
 			}
 		form.attrs.autocomplete = 
 			attribute autocomplete {
-				w:string "on" | w:string "off"
+				 "on" |  "off"
 			}
 		# REVISIT should this be case-insensitive in conforming XHTML documents?
 		form.attrs.enctype.data |= 
-			( w:string "text/plain" )
+			(  "text/plain" )
 
 ## Fieldset: <fieldset>, extensions
 
